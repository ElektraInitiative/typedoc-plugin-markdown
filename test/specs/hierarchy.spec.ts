import * as Handlebars from 'handlebars';
import { ProjectReflection } from 'typedoc';

describe(`Hierarchy:`, () => {
  let project: ProjectReflection;
  let helper: Handlebars.HelperDelegate;

  beforeAll(async () => {
    project = await global.bootstrap(['hierarchy.ts']);
    helper = Handlebars.helpers.hierarchy;
  });
  test(`should compile type hierarchy`, () => {
    const reflection = project.getChildByName('ParentClass');
    expect(
      global.compileHelper(helper, (reflection as any).typeHierarchy, 0),
    ).toMatchSnapshot();
  });

  test(`should compile nested type hierarchy`, () => {
    const reflection = project.getChildByName('ChildClassA');
    expect(
      global.compileHelper(helper, (reflection as any).typeHierarchy, 0),
    ).toMatchSnapshot();
  });
});
