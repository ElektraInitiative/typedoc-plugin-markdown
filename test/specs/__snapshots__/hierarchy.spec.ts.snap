// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Hierarchy: should compile nested type hierarchy 1`] = `
"- [\`ParentClass\`](ParentClass.md)

  ↳ **\`ChildClassA\`**

  ↳↳ [\`GrandChildClassA\`](GrandChildClassA.md)
"
`;

exports[`Hierarchy: should compile type hierarchy 1`] = `
"- **\`ParentClass\`**

  ↳ [\`ChildClassA\`](ChildClassA.md)

  ↳ [\`ChildClassB\`](ChildClassB.md)
"
`;
